src/Symfony/Component/Yaml/Inline.php

```diff
@@ -546,9 +547,8 @@ private static function parseMapping(string $mapping, int $flags, int &$i = 0, a
     private static function evaluateScalar(string $scalar, int $flags, array $references = [])
     {
         $scalar = trim($scalar);
-        $scalarLower = strtolower($scalar);
 
-        if (0 === strpos($scalar, '*')) {
+        if (0 === strncmp($scalar, '*', 1)) {
```

Comment:
One check and call to strncmp should be faster than two checks (one being a negation), at least with opcache. Benefit is that `strncmp` doesn't care if string is zero-length.

Reply:
https://3v4l.org/Zv1mO

Reply:
Thanks @gharlan - I originally hesitated to introduce null coalesce since I did not see any existing cases of this being used in the parser. It would perform better - should I spend the effort to use null coalesce as alternative to multiple conditions in this PR?

To illustrate `strncmp` in cases with longer strings, I made https://3v4l.org/HoeuA.

Null coalesce beats both. By a mile.